tests/test-ansible-functional.sh

```diff
@@ -154,7 +154,9 @@ function enable_maas_api {
   #                  different virtualenv which are done in three tasks. These three tasks are being
   #                  run in this particular order to ensure "pyrax" installs correctly.
   PYTHON_BIN="$(which python)"
-  virtualenv --python="${PYTHON_BIN}" /opt/test-maas
+  virtualenv --python="${PYTHON_BIN}" --no-pip /opt/test-maas
+  curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py
```

Comment:
get-pip.py may already exist on the host - add a check if it's already in /opt/get-pip.py before downloading it again, otherwise relying on an external resource may cause a failure

Also, there's a robust function for this in https://github.com/openstack/openstack-ansible/blob/master/scripts/scripts-library.sh#L355-L381 which you could either re-use, or copy and use. It does retries and also executes against a primary and fallback URL. You can also use that function to get the right version of pip/setuptools/wheel into the venv you're building here, something like this: https://github.com/openstack/openstack-ansible/blob/master/scripts/bootstrap-ansible.sh#L154-L160